GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -506,6 +513,35 @@ private static bool CheckConditionWithRetry(Func<bool> predicate, int retries, i
             return result;
         }
 
+        private EventMetadata GetProcessInteractionData(ref ConcurrentDictionary<string, PlaceHolderCreateCounter> collectedData, ref EventLevel eventLevel)
+        {
+            EventMetadata metadata = new EventMetadata();
+
+            if (collectedData.Count > 0)
+            {
+                ConcurrentDictionary<string, PlaceHolderCreateCounter> localData = collectedData;
+                collectedData = new ConcurrentDictionary<string, PlaceHolderCreateCounter>(StringComparer.OrdinalIgnoreCase);
+
+                int count = 0;
+                foreach (KeyValuePair<string, PlaceHolderCreateCounter> processCount in
+                    localData.OrderByDescending((KeyValuePair<string, PlaceHolderCreateCounter> kvp) => kvp.Value.Count))

```

Comment:
foreach (KeyValuePair<string, PlaceHolderCreateCounter> processCount in 
                    localData.OrderByDescending((KeyValuePair<string, PlaceHolderCreateCounter> kvp) => kvp.Value.Count).Take(10))

Then you don't need the count variable.